python/ray/data/_internal/numpy_support.py

```diff
@@ -45,28 +45,76 @@ def validate_numpy_batch(batch: Union[Dict[str, np.ndarray], Dict[str, list]]) -
 
 
 def _detect_highest_datetime_precision(datetime_list: List[datetime]) -> str:
+    """Detect the highest precision for a list of datetime objects.
```

Comment:
didn't know there is such a function. This looks nicer than the previous approach. 👍